token/services/owner/manager.go

```diff
@@ -0,0 +1,245 @@
+
+	v, err := net.Vault(tms.ID().Channel)
+	if err != nil {
+		return errors.Errorf("failed to get vault for [%s:%s:%s]", tms.ID().Network, tms.ID().Channel, tms.ID())
+	}
+	var pendingTXs []string
+	type ToBeUpdated struct {
+		TxID   string
+		Status ttxdb.TxStatus
+	}
+	var toBeUpdated []ToBeUpdated
+	for {
+		tr, err := it.Next()
+		if err != nil {
+			return errors.Errorf("failed to get next tx record for [%s:%s:%s]", tms.ID().Network, tms.ID().Channel, tms.ID())
+		}
+		if tr == nil {
+			break
+		}
+		if tr.Status == ttxdb.Pending {
+			logger.Debugf("found pending transaction [%s] at [%s:%s]", tr.TxID, tms.ID().Network, tms.ID().Channel)
+			found := false
+			for _, txID := range pendingTXs {
+				if tr.TxID == txID {
+					found = true
+					break
+				}
+			}
+			if found {
+				continue
+			}
+
+			// check the status of the pending transactions in the vault
+			status, err := v.Status(tr.TxID)
+			if err != nil {
+				pendingTXs = append(pendingTXs, tr.TxID)
+				continue
+			}
+
+			var txStatus ttxdb.TxStatus
+			switch status {
+			case network.Valid:
+				txStatus = ttxdb.Confirmed
+			case network.Invalid:
+				txStatus = ttxdb.Deleted
+			default:
+				pendingTXs = append(pendingTXs, tr.TxID)
+				continue
+			}
+			toBeUpdated = append(toBeUpdated, ToBeUpdated{
+				TxID:   tr.TxID,
+				Status: txStatus,
+			})
+		}
+	}
+	qe.Done()
+
+	for _, updated := range toBeUpdated {
+		if err := owner.db.SetStatus(updated.TxID, updated.Status); err != nil {
+			return errors.WithMessagef(err, "failed setting status for request %s", updated.TxID)
+		}
+		logger.Debugf("found transaction [%s] in vault with status [%d], corresponding pending transaction updated", updated.TxID, updated.Status)
+	}
+
+	logger.Debugf("ownerdb [%s:%s], found [%d] pending transactions", tms.ID().Network, tms.ID().Channel, len(pendingTXs))
+
+	for _, txID := range pendingTXs {
+		if err := net.SubscribeTxStatusChanges(txID, &TxStatusChangesListener{net, owner.db}); err != nil {
+			return errors.WithMessagef(err, "failed to subscribe event listener to network [%s:%s] for [%s]", tms.ID().Network, tms.ID().Channel, txID)
+		}
+	}
+
+	return nil
+}
+
+var (
+	managerType = reflect.TypeOf((*Manager)(nil))
+)
+
+// Get returns the Owner instance for the passed TMS
+func Get(sp view.ServiceProvider, tms *token.ManagementService) *Owner {
+	if tms == nil {
+		logger.Debugf("no TMS provided")
+		return nil
+	}
+	s, err := sp.GetService(managerType)
+	if err != nil {
+		logger.Errorf("failed to get manager service: [%s]", err)
+		return nil
+	}
+	auditor, err := s.(*Manager).Owner(tms)
+	if err != nil {
+		logger.Errorf("failed to get db for TMS [%s]: [%s]", tms.ID(), err)
+		return nil
+	}
+	return auditor
+}
+
+// New returns the Owner instance for the passed TMS
+func New(sp view.ServiceProvider, tms *token.ManagementService) *Owner {
+	return Get(sp, tms)
+}
+
+type tmsWallet struct {
+	tms *token.ManagementService
+}
+
+func (t *tmsWallet) ID() string {
+	return ""
```

Comment:
would tmsWallet always have empty string as ID?

Reply:
good point, let's set it to the TMS's id